Add search bar functionality

danhaj opened the pull request.

rafalcymerys commented:
@danhaj since you copied the standard VSF component, it's hard to see your changes here :) Could you describe briefly what was changed compared to the stock implementation?

danhaj commented:
@rafalcymerys Standard VSF `AppHeader` uses mocked products to show search results. I've added fetching products in AppHeader and changed a bit computed properties (`products` and `categories`) in SearchResults.
